### PR TITLE
Closes #4630: Clear out SyncEnginesStorage during logout

### LINKED ISSUE #4630: SyncEnginesStorage needs to be cleared while logging out

### PR DESCRIPTION
**Pull request by grigoryk**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/SyncEnginesStorage.kt [hunk 1]
```diff
@@ -46,5 +46,12 @@ class SyncEnginesStorage(private val context: Context) {
         storage().edit().putBoolean(engine.nativeName, status).apply()
     }
 
+    /**
+     * Clears out any stored [SyncEngine] state.
+     */
+    internal fun clear() {
+        storage().edit().clear().apply()
```
Comment (Amejia481) [inline components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/SyncEnginesStorage.kt]: Could we add a test for this ? :)

Comment (grigoryk) [reply components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/SyncEnginesStorage.kt]: There's a test in FxaAccountManager, but maybe an explicit one wouldn't hurt as well :)

**Comment by grigoryk** [comment]
bors r=Amejia481

File: components/service/firefox-accounts/src/test/java/mozilla/components/service/fxa/manager/SyncEnginesStorageTest.kt [hunk 2]
```diff
@@ -38,5 +39,8 @@ class SyncEnginesStorageTest {
                 SyncEngine.Other("test") to true
             ), store.getStatus()
         )
+
+        store.clear()
+        assertTrue(store.getStatus().isNullOrEmpty())
```
Comment (Amejia481) [inline components/service/firefox-accounts/src/test/java/mozilla/components/service/fxa/manager/SyncEnginesStorageTest.kt]: :)

**Pull request merged**